conformance/README.md

```diff
@@ -0,0 +1,51 @@
+
```

Comment:
Note that https://gateway-api-inference-extension.sigs.k8s.io/guides/implementers/#conformance-tests must link to this doc.

Reply:
@danehans  Done. I added the page as a separate doc and linked it in though it may make sense to add it directly to the implementers guide. kindly let me know what you think.